examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js

```diff
@@ -0,0 +1,655 @@
+// 02_simple_encoding_dash_cenc_hls_fairplay.js
+
+const Promise = require('bluebird');
+const Bitmovin = require('bitmovin-javascript').default;
+
+
+const BITMOVIN_API_KEY = '';
+const bitmovin = new Bitmovin({apiKey: BITMOVIN_API_KEY, debug: true});
+
+const INPUT_HTTP_HOST = '';
+const INPUT_HTTP_PATH = '';
+
+const OUTPUT_S3_ACCESS_KEY = '';
+const OUTPUT_S3_SECRET_KEY = '';
+const OUTPUT_S3_BUCKET_NAME = '';
+const OUTPUT_S3_BASE_PATH = '';
+
+const DRM_CENC_KEY = '';
+const DRM_CENC_KID = '';
+const DRM_CENC_WIDEVINE_PSSH = '';
+const DRM_CENC_PLAYREADY_LAURL = '';
+const DRM_CENC_PLAYREADY_PSSH = '';
+
+const DRM_FAIRPLAY_KEY = '';
+const DRM_FAIRPLAY_IV = '';
+const DRM_FAIRPLAY_URI = '';
+
+
+const main = () => {
+  return new Promise((resolve, reject) => {
+
+    const httpInputCreationPromise = createInput();
+    const s3OutputCreationPromise = createOutput();
+    const encodingResourceCreationPromise = createEncoding();
+    const codecConfigurationH264At816pPromise = createH264CodecConfiguration(1920, 816, 4800000, 24.0);
+    const codecConfigurationH264At544pPromise = createH264CodecConfiguration(1280, 544, 2400000, 24.0);
+    const codecConfigurationH264At272pPromise = createH264CodecConfiguration(640, 272, 1200000, 24.0);
+    const codecConfigurationAACPromise = createAACCodecConfiguration(128000, 48000);
+
+    Promise.all(
+      [
+        httpInputCreationPromise,
+        s3OutputCreationPromise,
+        encodingResourceCreationPromise,
+        codecConfigurationH264At816pPromise,
+        codecConfigurationH264At544pPromise,
+        codecConfigurationH264At272pPromise,
+        codecConfigurationAACPromise
+      ]
+    ).then(([
+        input,
+        output,
+        encoding,
+        codecConfigurationH264At816p,
+        codecConfigurationH264At544p,
+        codecConfigurationH264At272p,
+        codecConfigurationAAC
+      ]) => {
+      console.log('Successfully created input, output and codec configurations.');
+      const dashManifestCreationPromise = createDashManifestWithPeriodAndAdaptationSets(output.id);
+      const hlsManifestCreationPromise = createHlsManifest(output.id);
+      Promise.all([dashManifestCreationPromise, hlsManifestCreationPromise]).then(([dashManifest, hlsManifest]) => {
+        console.log('Successfully created DASH and HLS Manifest Resources');
+        [createdDashManifest, createdDashManifestPeriod, createdDashManifestAudioAdaptationSet, createdDashManifestVideoAdaptationSet] = dashManifest;
```

Comment:
In promises the order you add them to the `all([])` array defines the order you get the promise `.then` parameter array @lukaskroepfl